lti_consumer/models.py

```diff
@@ -534,9 +550,30 @@ def _get_lti_1p3_consumer(self):
                 tool_key=self.lti_1p3_tool_public_key,
                 tool_keyset_url=self.lti_1p3_tool_keyset_url,
             )
+        elif self.config_store == self.CONFIG_EXTERNAL:
+            block = compat.load_enough_xblock(self.location)
+            external = get_external_config_from_filter({}, self.external_id)
+
+            consumer = consumer_class(
+                iss=get_lti_api_base(),
+                # Use the xblock values if lti_1p3_oidc_url or lti_1p3_launch_url
+                # is not set on the external configuration.
+                lti_oidc_url=block.lti_1p3_oidc_url or external.get('lti_1p3_oidc_url'),
+                lti_launch_url=block.lti_1p3_launch_url or external.get('lti_1p3_launch_url'),
```

Comment:
We introduce it so course authors can override the OIDC URL, launch URL or deep linking launch URL, without the need to create a new configuration. We could add also the DB values as a fallback, I didn't wanted to introduce to much options on this fallback but also having the fallback on the DB might be useful. What do you suggest this be documented on?, I just thought about adding a code comment to describe this behavior but I agree it might be better to have it documented.

Reply:
@kuipumu , ah, right, we use them as overrides, not fallbacks.

Should we also override other values (like `client_id`, `rsa_key`) through the `LtiConfiguration`? _Quick context: I'm talking only about the `LtiConfiguration or ExternalConfiguration` condition since these values cannot be easily modified (or are not stored) within the XBlock fields._

I believe that the most discoverable solution would be adding this information directly to the help text of each XBlock/DB field that can be overridden this way. Something like _"If the configuration is retrieved from an external store, this value overrides the one specified in the external configuration."_, but with an indication that this is not applicable when you use the `CONFIG_ON_DB` option.

@Zacharis278, what do you think?

Reply:
I think an issue we run into with the override functionality is right now the configuration options are exclusive. You pick either xblock, database, or external. If you choose external it's ambiguous what the next most important configuration location would be. I'd agree making this not applicable `CONFIG_ON_DB` is probably the simplest short term solution around that. Another thing is that this would differ from the 1.1 implementation. If we're going to add an override feature its behavior should be consistent across LTI versions.

I'm leaning towards suggesting we break override behavior out into it's own PR and instead focus on parity with the existing 1.1 solution in here. Taking time to discuss/consider the full list of fields that are potentially overridable at the block level and how we visually communicate to course teams how it would work might hold up the core behavior of this PR.

Reply:
@Zacharis278 I agree that we should make this PR only about making LTI 1.3 external configuration 1:1 with the LTI 1.1 implementation and move other specific features (multi-tenancy, field override) into separate PRs. I already removed anything related to field overrides and multi-tenancy from the PR.